ui/src/shared/utils/preventNegativeZero.test.ts

```diff
@@ -0,0 +1,40 @@
+import {preventNegativeZero} from 'src/shared/utils/preventNegativeZero'
+
+describe('preventNegativeZero', () => {
+  it('should not alter non-zero numbers', () => {
+    expect(preventNegativeZero(1)).toEqual(1)
+    expect(preventNegativeZero(0.000000000001)).toEqual(0.000000000001)
+    expect(preventNegativeZero(-456.78)).toEqual(-456.78)
+
+    let nonZeroString = '0.000000000000000000000000001'
+    expect(preventNegativeZero(nonZeroString)).toEqual(nonZeroString)
+
+    nonZeroString = '1234567890'
+    expect(preventNegativeZero(nonZeroString)).toEqual(nonZeroString)
+
+    nonZeroString = '-123456789.0001'
+    expect(preventNegativeZero(nonZeroString)).toEqual(nonZeroString)
+  })
+  it('should handle negative zero as a number', () => {
+    expect(preventNegativeZero(-0)).toEqual(0)
+    expect(preventNegativeZero(-0.0)).toEqual(0)
+    /*
+      // Note: prettier prevents us from adding additional tests with these patterns:
```

Comment:
screw that razzmatazz. our tools are here to help us, not force us to conform to arbitrary patterns.

let's add these tests back and then add this to them

```
// prettier-ignore
```
so that prettier lets us do our thing.

https://prettier.io/docs/en/ignore.html#javascript